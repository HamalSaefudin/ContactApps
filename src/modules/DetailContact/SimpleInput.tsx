import Icon, {ICON_CATEGORY} from '@/components/Icons';
import Colors from '@/theme/Colors';
import Layout from '@/theme/Layout';
import React, {useRef} from 'react';
import {KeyboardTypeOptions, StyleSheet, TextInput, View} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import ShimmerPlaceholder from 'react-native-shimmer-placeholder';
import {moderateScale} from 'react-native-size-matters';

type Props = {
  value: string;
  onChangeText: (text: string, fieldName: string) => void;
  title: string;
  keyboardType?: KeyboardTypeOptions;
  isLoading?: boolean;
  fieldName: string;
};

function SimpleInput(props: Props) {
  const ref = useRef<TextInput>(null);
  return (
    <ShimmerPlaceholder
      visible={!props.isLoading}
      LinearGradient={LinearGradient}
      shimmerStyle={[Layout.fullWidth, {height: moderateScale(40)}]}>
      <TextInput
        ref={ref}
        value={props.value}
        keyboardType={props.keyboardType || 'default'}
        placeholder={props.title}
        onChangeText={val => props.onChangeText(val, props.fieldName)}
        style={styles.input}
      />
      <View style={styles.icEdit}>
        <Icon
          testID="icon-pencil"
          category={ICON_CATEGORY.OCTICONS}
          name="pencil"
        />
      </View>
    </ShimmerPlaceholder>
  );
}

const styles = StyleSheet.create({
  icEdit: {
    position: 'absolute',
    right: moderateScale(20),
    top: moderateScale(20),
  },
  input: {
    borderBottomWidth: 1,
    borderColor: Colors.common.border,
    paddingVertical: moderateScale(20),
  },
});

export default SimpleInput;
