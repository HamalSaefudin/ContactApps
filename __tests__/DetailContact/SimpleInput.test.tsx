import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import SimpleInput from '@/modules/DetailContact/SimpleInput';

describe('SimpleInput', () => {
  const mockOnChangeText = jest.fn();

  it('to match snapshot', () => {
    const {toJSON} = render(
      <SimpleInput
        value=""
        onChangeText={mockOnChangeText}
        title="First Name"
        isLoading={true}
        fieldName="testField"
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when not loading', () => {
    const {getByPlaceholderText, getByTestId} = render(
      <SimpleInput
        value="Hamal Saefudin"
        onChangeText={mockOnChangeText}
        title="First Name"
        isLoading={false}
        fieldName="firstName"
      />,
    );

    const input = getByPlaceholderText('First Name');
    expect(input.props.value).toBe('Hamal Saefudin');

    const icon = getByTestId('icon-pencil');
    expect(icon).toBeTruthy();
  });

  it('calls onChangeText', () => {
    const {getByPlaceholderText} = render(
      <SimpleInput
        value=""
        onChangeText={mockOnChangeText}
        title="First Name"
        isLoading={false}
        fieldName="firstName"
      />,
    );

    const input = getByPlaceholderText('First Name');
    fireEvent.changeText(input, 'New Value');
    expect(mockOnChangeText).toHaveBeenCalledWith('New Value', 'firstName');
  });
});
