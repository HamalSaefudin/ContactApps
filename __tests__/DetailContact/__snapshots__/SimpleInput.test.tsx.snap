// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleInput to match snapshot 1`] = `
<View
  style={
    [
      {
        "height": 15,
        "width": 200,
      },
      {
        "overflow": "hidden",
      },
      [
        {
          "width": "100%",
        },
        {
          "height": 62.857142857142854,
        },
      ],
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "height": 0,
          "opacity": 0,
          "width": 0,
        },
        false,
      ]
    }
  >
    <TextInput
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="First Name"
      style={
        {
          "borderBottomWidth": 1,
          "borderColor": "#D9D9D9",
          "paddingVertical": 31.428571428571427,
        }
      }
      value=""
    />
    <View
      style={
        {
          "position": "absolute",
          "right": 31.428571428571427,
          "top": 31.428571428571427,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#000000",
              "fontSize": 25.142857142857142,
            },
            undefined,
            {
              "fontFamily": "Octicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
        testID="icon-pencil"
      >
        Ôäê
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#ebebeb",
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "flex": 1,
          "transform": [
            {
              "translateX": -200,
            },
          ],
        }
      }
    >
      <BVLinearGradient
        colors={
          [
            4293651435,
            4291151301,
            4293651435,
          ]
        }
        endPoint={
          {
            "x": 2,
            "y": 0.5,
          }
        }
        locations={
          [
            0.3,
            0.5,
            0.7,
          ]
        }
        startPoint={
          {
            "x": -1,
            "y": 0.5,
          }
        }
        style={
          {
            "flex": 1,
            "width": 200,
          }
        }
      />
    </View>
  </View>
</View>
`;
